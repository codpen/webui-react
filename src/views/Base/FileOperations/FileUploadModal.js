import React, {useState} from 'react';
import {
    Button,
    Col,
    Container,
    FormGroup,
    Input,
    Modal,
    ModalBody,
    ModalFooter,
    ModalHeader,
    Row,
    UncontrolledTooltip
} from 'reactstrap';
import axiosInstance from "../../../utils/API/API";
import {addColonAtLast} from "../../../utils/Tools";
import {toast} from "react-toastify";
import * as RclonePropTypes from "../../../utils/RclonePropTypes";

/**
 * New Mount Modal shows a button for opening a modal for new mount and then executes okHandle when positive
 * button is clicked
 * @param props
 * @returns {*}
 * @constructor
 */
const FileUploadModal = (props) => {
    const {
        currentPath
    } = props;

    const [modal, setModal] = useState(false);


    const [files, setFiles] = useState(null);


    const toggle = () => setModal(!modal);


    const changeFilesHandler = (e) => {
        setFiles(e.target.files);
    }

    const isUploadDisabled = () => files == null;


    const fileUploadHandler = (_) => {
        const data = new FormData();
        console.log(files);
        for (let i = 0; i < files.length; i++) {
            // get item
            let file = files.item(i);

            data.append("file" + i, file);
        }


        axiosInstance.post(`operations/uploadfile?fs=` +
            `${addColonAtLast(currentPath.remoteName)}` +
            `&remote=${currentPath.remotePath}`, data)
            .then(res => {
                toast.info("File uploaded successfully");
            }, err => {
                toast.error("File upload failed");
            });
    }

    return (
        <div data-test="fileUploadModalComponent">
            <Button className="btn-explorer-action" id="UploadButton"
                    onClick={toggle}>
                <i className="fa fa-lg fa-upload"/>
            </Button>
            <UncontrolledTooltip placement="right" target="UploadButton">
                Upload file(s)
            </UncontrolledTooltip>
            <Modal isOpen={modal} toggle={toggle}>
                <ModalHeader toggle={toggle}>New Mount</ModalHeader>
                <ModalBody>
                    <Container fluid>
                        <Row className="upload-box">
                            <Col sm={3}/>
                            <Col sm={6}>

                                <label htmlFor="upload-file">
                                    <Row>
                                        <i className="fa fa-lg fa-plus mx-auto mt-5"/>
                                        <p className="text-center mt-2">Click here or drag and drop files to upload</p>
                                    </Row>

                                </label>
                            </Col>
                            <Col sm={3}/>
                        </Row>
                        <Row>
                            <FormGroup row className="pl-2 pr-2">
                                <Input id="upload-file" type="file" name="file" onChange={changeFilesHandler} multiple/>
                            </FormGroup>
                        </Row>
                    </Container>
                </ModalBody>
                <ModalFooter>
                    <Button data-test="ok-button" color="primary" onClick={fileUploadHandler}
                            disabled={isUploadDisabled()}>Create</Button>{' '}
                    <Button data-test="cancel-button" color="secondary" onClick={toggle}>Cancel</Button>
                </ModalFooter>
            </Modal>
        </div>
    );
}

FileUploadModal.propTypes = {
    currentPath: RclonePropTypes.PROP_CURRENT_PATH
}

export default FileUploadModal;