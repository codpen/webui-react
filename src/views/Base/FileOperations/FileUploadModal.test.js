import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import FileUploadModal from "./FileUploadModal";
import {findByTestAttr, testStore} from "../../../../Utils";


const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);

    const component = shallow(<FileUploadModal {...props} store={store}/>);
    return component;
};


describe('File upload Modal', function () {
    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {
                currentPath: {
                    remoteName: "local",
                    remotePath: ""
                }
            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAttr(wrapper, "fileUploadModalComponent");
            expect(component).toHaveLength(1);
        });

        it('should have ok and cancel buttons', function () {
            const okButton = findByTestAttr(wrapper, "ok-button")
            expect(okButton).toHaveLength(1);

            const cancelButton = findByTestAttr(wrapper, "cancel-button");
            expect(cancelButton).toHaveLength(1);
        })


        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

    });
});
