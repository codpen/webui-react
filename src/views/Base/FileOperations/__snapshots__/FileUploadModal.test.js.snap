// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File upload Modal renders should match snapshot 1`] = `
<div
  data-test="fileUploadModalComponent"
>
  <Button
    className="btn-explorer-action"
    color="secondary"
    id="UploadButton"
    onClick={[Function]}
    tag="button"
  >
    <i
      className="fa fa-lg fa-upload"
    />
  </Button>
  <UncontrolledTooltip
    placement="right"
    target="UploadButton"
  >
    Upload file(s)
  </UncontrolledTooltip>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      New Mount
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <Container
        fluid={true}
        tag="div"
      >
        <Row
          className="upload-box"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Col
            sm={3}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          />
          <Col
            sm={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <label
              htmlFor="upload-file"
            >
              <Row
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <i
                  className="fa fa-lg fa-plus mx-auto mt-5"
                />
                <p
                  className="text-center mt-2"
                >
                  Click here or drag and drop files to upload
                </p>
              </Row>
            </label>
          </Col>
          <Col
            sm={3}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          />
        </Row>
        <Row
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <FormGroup
            className="pl-2 pr-2"
            row={true}
            tag="div"
          >
            <Input
              id="upload-file"
              multiple={true}
              name="file"
              onChange={[Function]}
              type="file"
            />
          </FormGroup>
        </Row>
      </Container>
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        data-test="ok-button"
        disabled={true}
        onClick={[Function]}
        tag="button"
      >
        Create
      </Button>
       
      <Button
        color="secondary"
        data-test="cancel-button"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
